web/client/components/resources/modals/fragments/DetailsRow.jsx

```diff
@@ -1,85 +1,94 @@
-const React = require('react');
-const {Row, Col} = require('react-bootstrap');
-const Spinner = require('react-spinkit');
-const { isNil } = require('lodash');
-const Toolbar = require('../../../misc/toolbar/Toolbar');
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
 
-const Message = require('../../../I18N/Message');
-const { NO_DETAILS_AVAILABLE } = require('../../../../actions/maps');
+import React from 'react';
+import { Row, Col } from 'react-bootstrap';
+import Spinner from 'react-spinkit';
+import { isNil } from 'lodash';
+import Toolbar from '../../../misc/toolbar/Toolbar';
 
+import Message from '../../../I18N/Message';
 
-module.exports = ({
-    saving,
-    hideGroupProperties,
+
+export default ({
+    resource = {},
+    showPreview = false,
     editDetailsDisabled,
     detailsText,
-    detailsBackup,
-    onToggleGroupProperties = () => { },
-    onUndoDetails = () => { },
-    onToggleDetailsSheet = () => { },
-    onUpdateDetails = () => { }
+    canUndo = false,
+    onShowPreview = () => {},
+    onHidePreview = () => {},
+    onUndo = () => {},
+    onShowDetailsSheet = () => {},
+    onUpdate = () => {},
+    onDelete = () => {}
 }) => {
     return (
-        <div className={"ms-section" + (hideGroupProperties ? ' ms-transition' : '')}>
+        <div className={"ms-section" + (showPreview ? ' ms-transition' : '')}>
             <div className="mapstore-block-width">
                 <Row>
                     <Col xs={6}>
                         <div className="m-label">
-                            {detailsText === "" ? <Message msgId="map.details.add" /> : <Message msgId="map.details.rowTitle" />}
+                            {detailsText === 'NODATA' ? <Message msgId="map.details.add" /> : <Message msgId="map.details.rowTitle" />}
```

Comment:
can you handle 'NODATA' at an upper level? NODATA is unfortunately a workaround for missing back-end functionalities.

"NODATA" or detail attribute not present represent the same "no detail card" condition. 
You should isolate this at an upper level in order to avoid to duplicate "NODATA" here and there, with the possibility of generatin bugs. For instance, here if detailsText is "" or undefined, what happens?
Please isolate this passing instead an empty resource